devito/ops/node_factory.py

```diff
@@ -68,14 +68,25 @@ def new_ops_arg(self, indexed, is_write):
 
         return OpsAccess(symbol_to_access, space_indices)
 
-    def new_ops_gbl(self, c):
-        if c in self.ops_args:
-            return self.ops_args[c].accessible
+    def new_ops_gbl(self, expr):
+        """
+        Ensure Constant symbols are accessed as C pointers.
 
-        new_c = AccessibleInfo(Constant(name='*%s' % c.name, dtype=c.dtype),
-                               None,
-                               None)
-        self.ops_args[c] = new_c
-        self.ops_params.append(new_c.accessible)
+        Parameters
+        ----------
+        expr : Expression
+               Constant expression.
 
-        return new_c.accessible
+        Returns
+        -------
+        Constant
+            Constant symbol, as a pointer.
+        """
+        if expr not in self.ops_args:
+            self.ops_args[expr] = AccessibleInfo(Constant(name='*%s' % expr.name,
```

Comment:
hack

Reply:
is it so because there is a `*` in the `name` of a `Constant`?

Reply:
yes

you should try to figure how devito manages to generate all those stars